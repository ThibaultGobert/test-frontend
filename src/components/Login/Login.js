import React from 'react';
import PropTypes from 'prop-types';
import { Form, Input, Icon } from 'semantic-ui-react';

import CodeTag from '../CodeTag';
import Loader from '../shared/Loader';
import ErrorMessage from '../shared/ErrorMessage';

import './Login.css';
import ramdassWelcome from '../../assets/images/login/ramdass-welkom.png';

const Login = ({
  impersonate,
  credentials,
  hidden,
  error,
  loading,
  submit,
  onChange,
  toggleHidden,
  toggleImpersonate
}) => {
  return (
    <div className="Login">
      <img className="Login__Rambdass" src={ramdassWelcome} alt=""/>
      <Loader active={loading} message="Even geduld. Wij proberen u in te loggen..." />

      <div className="Login__Wrapper">
        <Form className="Login__Form" onSubmit={submit}>
          <CodeTag text="Start" />
          <Form.Group inline>
            <Form.Field>
              <CodeTag text="Uw gebruikersnaam" className="Login__CodeTag--highlight" closing />
              <input
                placeholder="Uw gebruikersnaam"
                onChange={onChange}
                name="username"
                value={credentials.username}
              />
            </Form.Field>
          </Form.Group>

          {impersonate && (
            <Form.Group inline>
              <Form.Field>
                <CodeTag text="Gebruikersnaam kind" className="Login__CodeTag--highlight" closing />
                <input
                  placeholder="Gebruikersnaam kind"
                  onChange={onChange}
                  name="child_username"
                />
              </Form.Field>
            </Form.Group>
          )}

          <Form.Group inline>
            <Form.Field>
              <CodeTag text="Wachtwoord" className="Login__CodeTag--highlight" closing />
              <Input
                type={hidden ? 'password' : 'text'}
                placeholder="Wachtwoord"
                onChange={onChange}
                name="password"
                value={credentials.password}
                icon={
                  !impersonate && (
                    <Icon name="eye" className="eye-icon" inverted link onClick={toggleHidden} />
                  )
                }
              />
            </Form.Field>
          </Form.Group>
          <CodeTag text="Start" closing />
          {error && <ErrorMessage header="Login mislukt" message={error} />}
          <Form.Button primary>Login</Form.Button>
        </Form>
      </div>

      <button className="Login__ImpersonateLink" onClick={toggleImpersonate}>
        <Icon name="lock" />
      </button>
    </div>
  );
};

Login.propTypes = {
  impersonate: PropTypes.bool,
  credentials: PropTypes.object,
  hidden: PropTypes.bool,
  hasError: PropTypes.bool,
  loading: PropTypes.bool
};

export default Login;
