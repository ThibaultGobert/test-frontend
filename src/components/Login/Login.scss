.Login .Login__Wrapper {
  background: #337ab7;
  display: inline-block;
  margin-left: 250px;
  margin-top: 50px;

  padding: 2em;
  border-radius: 1em;
}

.Login .Login__Form {
  min-width: 750px;
  margin: 0 auto;

  padding: 5em 3em;
  background: #262626;
  border-radius: 25px;
  border: 10px solid #222;
}

.Login .Login__CodeTag--highlight {
  color: yellow;
  margin-left: 1em !important;
  min-width: 12em;
  display: inline-block;
}

.Login .field input {
  width: 20em !important;
}

.Login .ui.form .field > label {
  color: #fff;
}

.Login button {
  float: right;
  margin-top: 1.5em !important;
}

.Login .Login__ImpersonateLink {
  position: absolute;
  right: 10px;
  bottom: 10px;
}

.Login .Login__Rambdass {
  position: absolute;
  bottom: 50px;
  left: 50px;
  width: 250px;
}
