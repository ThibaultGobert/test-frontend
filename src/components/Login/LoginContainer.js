import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';
import toastr from 'toastr';

import Login from './Login';
import api from '../../api/auth';
import * as roles from '../../constants/roles';

class LoginContainer extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      credentials: {
        username: '',
        password: '',
        child_username: ''
      },
      hidden: true,
      impersonate: false
    };

    this.onChange = this.onChange.bind(this);
    this.submit = this.submit.bind(this);
    this.toggleHidden = this.toggleHidden.bind(this);
    this.toggleImpersonate = this.toggleImpersonate.bind(this);
  }

  componentWillReceiveProps({ loggedIn }) {
    console.log(loggedIn);

    if (!isEmpty(loggedIn)) {
      if (loggedIn.role === roles.STUDENT_ROLE) {
        this.context.router.push('/studentprofile/clan');
      } else if (loggedIn.role === roles.TEACHER_ROLE) {
        this.context.router.push('/teacherprofile/overview');
      } else if (loggedIn.role === roles.WORKSHOP_STUDENT_ROLE) {
        this.context.router.push('/workshopprofile/overview');
      } else if (loggedIn.role === roles.EDITOR_ROLE) {
        this.context.router.push('/editorprofile/overview');
      } else if (loggedIn.role === roles.ADMIN_ROLE) {
        toastr.error('Geen toegang voor admins');
        this.props.actions.logOut();
      }
    }
  }

  submit(event) {
    event.preventDefault();

    const { loginStart, loginSuccess, loginError } = this.props.actions;


    loginStart();

    api
      .login(this.state.credentials)
      .then(loginSuccess)
      .catch(loginError);
  }

  onChange({ target }) {
    this.setState(prevState => ({
      credentials: {
        ...prevState.credentials,
        [target.name]: target.value
      }
    }));
  }

  toggleHidden() {
    this.setState(prevState => ({
      hidden: !prevState.hidden
    }));
  }

  toggleImpersonate() {
    this.setState(prevState => ({
      impersonate: !prevState.impersonate,
      hidden: true,
      credentials: { ...prevState.credentials, child_username: '' }
    }));
  }

  render() {
    const { error, loading } = this.props;

    return (
      <Login
        error={error}
        loading={loading}
        submit={this.submit}
        onChange={this.onChange}
        toggleHidden={this.toggleHidden}
        toggleImpersonate={this.toggleImpersonate}

        {...this.state}
      />
    );
  }
}

LoginContainer.propTypes = {
  loggedIn: PropTypes.object.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.object,
  actions: PropTypes.object
};

LoginContainer.contextTypes = {
  router: PropTypes.object
};

export default LoginContainer;
