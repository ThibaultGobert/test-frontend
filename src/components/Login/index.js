import { connect } from 'react-redux';
import mapActionCreatorsToProps from '../../functions/mapActionCreatorsToProps';
import { loginStart, loginSuccess, loginError, logOut } from '../../actions/auth';
import LoginContainer from './LoginContainer';

const mapStateToProps = state => ({
  loggedIn: state.loggedIn.data,
  loading: state.loggedIn.loading,
  error: state.loggedIn.error
});

const actionCreators = mapActionCreatorsToProps({
  loginStart,
  loginSuccess,
  loginError,
  logOut
});

export default connect(mapStateToProps, actionCreators)(LoginContainer);
