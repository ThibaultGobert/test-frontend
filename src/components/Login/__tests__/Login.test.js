import React from 'react';
import { mount, mountToJson, toJson } from '../../../utils/test/mount';

import Login from '../Login';

describe('Login', () => {
  it('should render Login', () => {
    const component = mountToJson(<Login credentials={{}} />);

    expect(component).toMatchSnapshot();
  });

  it('should render Login form with impersonate', () => {
    const component = mountToJson(<Login credentials={{}} impersonate />);

    expect(component).toMatchSnapshot();
  });

  it('should render Login with plain text password', () => {
    const component = mountToJson(<Login credentials={{}} hidden={false} />);

    expect(component).toMatchSnapshot();
  });

  it('should show error when hasError', () => {
    const component = mountToJson(<Login credentials={{}} hasError error="We failed!" />);

    expect(component).toMatchSnapshot();
  });

  it('should show loading screen when loading', () => {
    const component = mountToJson(<Login credentials={{}} loading />);

    expect(component).toMatchSnapshot();
  });
});
