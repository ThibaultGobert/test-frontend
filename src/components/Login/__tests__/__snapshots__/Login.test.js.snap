// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render Login 1`] = `
<Login
  credentials={Object {}}
>
  <div
    className="Login"
  >
    <img
      alt=""
      className="Login__Rambdass"
      src="ramdass-welkom.png"
    />
    <Loader
      message="Even geduld. Wij proberen u in te loggen..."
    >
      <Dimmer
        active={false}
      >
        <div
          className="ui dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="center"
            >
              <Loader
                size="medium"
              >
                <div
                  className="ui medium text loader"
                >
                  Even geduld. Wij proberen u in te loggen...
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </Dimmer>
    </Loader>
    <div
      className="Login__Wrapper"
    >
      <Form
        as="form"
        className="Login__Form"
      >
        <form
          className="ui form Login__Form"
          onSubmit={[Function]}
        >
          <CodeTag
            close={false}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Uw gebruikersnaam"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Uw gebruikersnaam
                      &gt;
                    </div>
                  </CodeTag>
                  <input
                    name="username"
                    placeholder="Uw gebruikersnaam"
                  />
                </div>
              </FormField>
            </div>
          </FormGroup>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Wachtwoord"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Wachtwoord
                      &gt;
                    </div>
                  </CodeTag>
                  <Input
                    icon={
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                        onClick={undefined}
                      />
                    }
                    name="password"
                    placeholder="Wachtwoord"
                    type="text"
                  >
                    <div
                      className="ui icon input"
                    >
                      <input
                        key="text"
                        name="password"
                        onChange={[Function]}
                        placeholder="Wachtwoord"
                        type="text"
                      />
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                      >
                        <i
                          aria-hidden="true"
                          className="eye inverted link icon eye-icon"
                        />
                      </Icon>
                    </div>
                  </Input>
                </div>
              </FormField>
            </div>
          </FormGroup>
          <CodeTag
            close={false}
            closing={true}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormButton
            as={[Function]}
            control={[Function]}
            primary={true}
          >
            <FormField
              control={[Function]}
              primary={true}
            >
              <div
                className="field"
              >
                <Button
                  as="button"
                  primary={true}
                >
                  <button
                    className="ui primary button"
                    onClick={[Function]}
                    role="button"
                  >
                    Login
                  </button>
                </Button>
              </div>
            </FormField>
          </FormButton>
        </form>
      </Form>
    </div>
    <button
      className="Login__ImpersonateLink"
    >
      <Icon
        as="i"
        name="lock"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </Icon>
    </button>
  </div>
</Login>
`;

exports[`Login should render Login form with impersonate 1`] = `
<Login
  credentials={Object {}}
  impersonate={true}
>
  <div
    className="Login"
  >
    <img
      alt=""
      className="Login__Rambdass"
      src="ramdass-welkom.png"
    />
    <Loader
      message="Even geduld. Wij proberen u in te loggen..."
    >
      <Dimmer
        active={false}
      >
        <div
          className="ui dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="center"
            >
              <Loader
                size="medium"
              >
                <div
                  className="ui medium text loader"
                >
                  Even geduld. Wij proberen u in te loggen...
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </Dimmer>
    </Loader>
    <div
      className="Login__Wrapper"
    >
      <Form
        as="form"
        className="Login__Form"
      >
        <form
          className="ui form Login__Form"
          onSubmit={[Function]}
        >
          <CodeTag
            close={false}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Uw gebruikersnaam"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Uw gebruikersnaam
                      &gt;
                    </div>
                  </CodeTag>
                  <input
                    name="username"
                    placeholder="Uw gebruikersnaam"
                  />
                </div>
              </FormField>
            </div>
          </FormGroup>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Gebruikersnaam kind"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Gebruikersnaam kind
                      &gt;
                    </div>
                  </CodeTag>
                  <input
                    name="child_username"
                    placeholder="Gebruikersnaam kind"
                  />
                </div>
              </FormField>
            </div>
          </FormGroup>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Wachtwoord"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Wachtwoord
                      &gt;
                    </div>
                  </CodeTag>
                  <Input
                    icon={false}
                    name="password"
                    placeholder="Wachtwoord"
                    type="text"
                  >
                    <div
                      className="ui input"
                    >
                      <input
                        key="text"
                        name="password"
                        onChange={[Function]}
                        placeholder="Wachtwoord"
                        type="text"
                      />
                    </div>
                  </Input>
                </div>
              </FormField>
            </div>
          </FormGroup>
          <CodeTag
            close={false}
            closing={true}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormButton
            as={[Function]}
            control={[Function]}
            primary={true}
          >
            <FormField
              control={[Function]}
              primary={true}
            >
              <div
                className="field"
              >
                <Button
                  as="button"
                  primary={true}
                >
                  <button
                    className="ui primary button"
                    onClick={[Function]}
                    role="button"
                  >
                    Login
                  </button>
                </Button>
              </div>
            </FormField>
          </FormButton>
        </form>
      </Form>
    </div>
    <button
      className="Login__ImpersonateLink"
    >
      <Icon
        as="i"
        name="lock"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </Icon>
    </button>
  </div>
</Login>
`;

exports[`Login should render Login with plain text password 1`] = `
<Login
  credentials={Object {}}
  hidden={false}
>
  <div
    className="Login"
  >
    <img
      alt=""
      className="Login__Rambdass"
      src="ramdass-welkom.png"
    />
    <Loader
      message="Even geduld. Wij proberen u in te loggen..."
    >
      <Dimmer
        active={false}
      >
        <div
          className="ui dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="center"
            >
              <Loader
                size="medium"
              >
                <div
                  className="ui medium text loader"
                >
                  Even geduld. Wij proberen u in te loggen...
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </Dimmer>
    </Loader>
    <div
      className="Login__Wrapper"
    >
      <Form
        as="form"
        className="Login__Form"
      >
        <form
          className="ui form Login__Form"
          onSubmit={[Function]}
        >
          <CodeTag
            close={false}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Uw gebruikersnaam"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Uw gebruikersnaam
                      &gt;
                    </div>
                  </CodeTag>
                  <input
                    name="username"
                    placeholder="Uw gebruikersnaam"
                  />
                </div>
              </FormField>
            </div>
          </FormGroup>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Wachtwoord"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Wachtwoord
                      &gt;
                    </div>
                  </CodeTag>
                  <Input
                    icon={
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                        onClick={undefined}
                      />
                    }
                    name="password"
                    placeholder="Wachtwoord"
                    type="text"
                  >
                    <div
                      className="ui icon input"
                    >
                      <input
                        key="text"
                        name="password"
                        onChange={[Function]}
                        placeholder="Wachtwoord"
                        type="text"
                      />
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                      >
                        <i
                          aria-hidden="true"
                          className="eye inverted link icon eye-icon"
                        />
                      </Icon>
                    </div>
                  </Input>
                </div>
              </FormField>
            </div>
          </FormGroup>
          <CodeTag
            close={false}
            closing={true}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormButton
            as={[Function]}
            control={[Function]}
            primary={true}
          >
            <FormField
              control={[Function]}
              primary={true}
            >
              <div
                className="field"
              >
                <Button
                  as="button"
                  primary={true}
                >
                  <button
                    className="ui primary button"
                    onClick={[Function]}
                    role="button"
                  >
                    Login
                  </button>
                </Button>
              </div>
            </FormField>
          </FormButton>
        </form>
      </Form>
    </div>
    <button
      className="Login__ImpersonateLink"
    >
      <Icon
        as="i"
        name="lock"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </Icon>
    </button>
  </div>
</Login>
`;

exports[`Login should show error when hasError 1`] = `
<Login
  credentials={Object {}}
  error="We failed!"
  hasError={true}
>
  <div
    className="Login"
  >
    <img
      alt=""
      className="Login__Rambdass"
      src="ramdass-welkom.png"
    />
    <Loader
      message="Even geduld. Wij proberen u in te loggen..."
    >
      <Dimmer
        active={false}
      >
        <div
          className="ui dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="center"
            >
              <Loader
                size="medium"
              >
                <div
                  className="ui medium text loader"
                >
                  Even geduld. Wij proberen u in te loggen...
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </Dimmer>
    </Loader>
    <div
      className="Login__Wrapper"
    >
      <Form
        as="form"
        className="Login__Form"
      >
        <form
          className="ui form Login__Form"
          onSubmit={[Function]}
        >
          <CodeTag
            close={false}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Uw gebruikersnaam"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Uw gebruikersnaam
                      &gt;
                    </div>
                  </CodeTag>
                  <input
                    name="username"
                    placeholder="Uw gebruikersnaam"
                  />
                </div>
              </FormField>
            </div>
          </FormGroup>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Wachtwoord"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Wachtwoord
                      &gt;
                    </div>
                  </CodeTag>
                  <Input
                    icon={
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                        onClick={undefined}
                      />
                    }
                    name="password"
                    placeholder="Wachtwoord"
                    type="text"
                  >
                    <div
                      className="ui icon input"
                    >
                      <input
                        key="text"
                        name="password"
                        onChange={[Function]}
                        placeholder="Wachtwoord"
                        type="text"
                      />
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                      >
                        <i
                          aria-hidden="true"
                          className="eye inverted link icon eye-icon"
                        />
                      </Icon>
                    </div>
                  </Input>
                </div>
              </FormField>
            </div>
          </FormGroup>
          <CodeTag
            close={false}
            closing={true}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <ErrorMessage
            header="Login mislukt"
            message="We failed!"
          >
            <Message
              negative={true}
            >
              <div
                className="ui negative message"
              >
                <MessageHeader>
                  <div
                    className="header"
                  >
                    Login mislukt
                  </div>
                </MessageHeader>
                <p>
                  We failed!
                </p>
              </div>
            </Message>
          </ErrorMessage>
          <FormButton
            as={[Function]}
            control={[Function]}
            primary={true}
          >
            <FormField
              control={[Function]}
              primary={true}
            >
              <div
                className="field"
              >
                <Button
                  as="button"
                  primary={true}
                >
                  <button
                    className="ui primary button"
                    onClick={[Function]}
                    role="button"
                  >
                    Login
                  </button>
                </Button>
              </div>
            </FormField>
          </FormButton>
        </form>
      </Form>
    </div>
    <button
      className="Login__ImpersonateLink"
    >
      <Icon
        as="i"
        name="lock"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </Icon>
    </button>
  </div>
</Login>
`;

exports[`Login should show loading screen when loading 1`] = `
<Login
  credentials={Object {}}
  loading={true}
>
  <div
    className="Login"
  >
    <img
      alt=""
      className="Login__Rambdass"
      src="ramdass-welkom.png"
    />
    <Loader
      active={true}
      message="Even geduld. Wij proberen u in te loggen..."
    >
      <Dimmer
        active={true}
      >
        <div
          className="ui active transition visible dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="center"
            >
              <Loader
                size="medium"
              >
                <div
                  className="ui medium text loader"
                >
                  Even geduld. Wij proberen u in te loggen...
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </Dimmer>
    </Loader>
    <div
      className="Login__Wrapper"
    >
      <Form
        as="form"
        className="Login__Form"
      >
        <form
          className="ui form Login__Form"
          onSubmit={[Function]}
        >
          <CodeTag
            close={false}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Uw gebruikersnaam"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Uw gebruikersnaam
                      &gt;
                    </div>
                  </CodeTag>
                  <input
                    name="username"
                    placeholder="Uw gebruikersnaam"
                  />
                </div>
              </FormField>
            </div>
          </FormGroup>
          <FormGroup
            inline={true}
          >
            <div
              className="inline fields"
            >
              <FormField>
                <div
                  className="field"
                >
                  <CodeTag
                    className="Login__CodeTag--highlight"
                    close={false}
                    closing={true}
                    text="Wachtwoord"
                  >
                    <div
                      className="CodeTag Login__CodeTag--highlight"
                    >
                      &lt;
                      Wachtwoord
                      &gt;
                    </div>
                  </CodeTag>
                  <Input
                    icon={
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                        onClick={undefined}
                      />
                    }
                    name="password"
                    placeholder="Wachtwoord"
                    type="text"
                  >
                    <div
                      className="ui icon input"
                    >
                      <input
                        key="text"
                        name="password"
                        onChange={[Function]}
                        placeholder="Wachtwoord"
                        type="text"
                      />
                      <Icon
                        as="i"
                        className="eye-icon"
                        inverted={true}
                        link={true}
                        name="eye"
                      >
                        <i
                          aria-hidden="true"
                          className="eye inverted link icon eye-icon"
                        />
                      </Icon>
                    </div>
                  </Input>
                </div>
              </FormField>
            </div>
          </FormGroup>
          <CodeTag
            close={false}
            closing={true}
            text="Start"
          >
            <div
              className="CodeTag"
            >
              &lt;
              Start
              &gt;
            </div>
          </CodeTag>
          <FormButton
            as={[Function]}
            control={[Function]}
            primary={true}
          >
            <FormField
              control={[Function]}
              primary={true}
            >
              <div
                className="field"
              >
                <Button
                  as="button"
                  primary={true}
                >
                  <button
                    className="ui primary button"
                    onClick={[Function]}
                    role="button"
                  >
                    Login
                  </button>
                </Button>
              </div>
            </FormField>
          </FormButton>
        </form>
      </Form>
    </div>
    <button
      className="Login__ImpersonateLink"
    >
      <Icon
        as="i"
        name="lock"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </Icon>
    </button>
  </div>
</Login>
`;
